src/app/bridge/bridgeProxy.ts

```diff
@@ -0,0 +1,160 @@
+import { Server as SocketIOServer, Socket } from 'socket.io';
+import type { Telemetry, Session, DashboardLayout } from '@irdashies/types';
+import type { IrSdkBridge, DashboardBridge } from '@irdashies/types';
+
+// Export current state so it can be accessed by other parts of the app
+export let currentDashboard: DashboardLayout | null = null;
+
+/**
+ * Bridge proxy that exposes Electron bridge data via WebSocket
+ * Allows external browsers to receive real-time telemetry, session data, and dashboard configuration
+ */
+export function createBridgeProxy(
+  httpServer: HTTPServer,
+  irsdkBridge: IrSdkBridge,
+  dashboardBridge?: DashboardBridge,
+  initialDemoMode = false
+) {
+  console.log('🚀 createBridgeProxy called!');
+  console.log('  Has irsdkBridge?', !!irsdkBridge);
+  console.log('  Has dashboardBridge?', !!dashboardBridge);
+  console.log('  Initial demo mode:', initialDemoMode);
+  
+  const io = new SocketIOServer(httpServer, {
+    cors: {
+      origin: '*',
+      methods: ['GET', 'POST'],
+    },
+  });
+
+  // Track connected clients
+  let currentTelemetry: Telemetry | null = null;
+  let currentSession: Session | null = null;
+  let isRunning = false;
+  let isDemoMode = initialDemoMode;
+
+  console.log('🎭 Bridge proxy initialized with demo mode:', isDemoMode);
+
+  console.log('🔍 Bridge proxy: Registering event listeners...');
+  console.log('🔍 Bridge has onTelemetry?', typeof irsdkBridge.onTelemetry);
+  console.log('🔍 Bridge has onSessionData?', typeof irsdkBridge.onSessionData);
+  console.log('🔍 Bridge has onRunningState?', typeof irsdkBridge.onRunningState);
+
+  // Function to subscribe to bridge events
+  const subscribeToBridge = (bridge: IrSdkBridge) => {
+    console.log('🔌 Bridge proxy: Subscribing to bridge events...');
+    
+    bridge.onTelemetry((telemetry: Telemetry) => {
+      currentTelemetry = telemetry;
+      console.log('🔄 Bridge proxy: Received telemetry, broadcasting to', io.engine.clientsCount, 'clients');
+      io.emit('telemetry', telemetry);
+    });
+    console.log('✅ Bridge proxy: onTelemetry listener registered');
+
+    bridge.onSessionData((session: Session) => {
+      currentSession = session;
+      console.log('🔄 Bridge proxy: Received session data, broadcasting to', io.engine.clientsCount, 'clients');
+      io.emit('sessionData', session);
+    });
+    console.log('✅ Bridge proxy: onSessionData listener registered');
+
+    bridge.onRunningState((running: boolean) => {
+      isRunning = running;
+      console.log('🔄 Bridge proxy: Received running state:', running, ', broadcasting to', io.engine.clientsCount, 'clients');
+      io.emit('runningState', running);
+    });
+    console.log('✅ Bridge proxy: onRunningState listener registered');
+  };
+
+  // Subscribe to initial bridge
+  subscribeToBridge(irsdkBridge);
+
+  // Return a function to resubscribe when bridge changes
+  const resubscribeToBridge = (newBridge: IrSdkBridge) => {
```

Comment:
what would be the scenario that causes resubscribeToBridge to be called? The main issue here is I think you might have a memory leak where subscribeToBridge is setting up event listeners but they never clear so if this runs again it will create duplicate listners.